plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,310 @@
+     */
+    protected abstract String stringInLogExpected();
+
+    /**
+     * 1.  Configure the instance with the {@link #configResource()} implemented.
+     * 2.  Check it was configured correctly.
+     * 3.  Export the full Jenkins configuration via Web UI (commented as it fails with maven defaultProperties).
+     * 4.  Export the schema via Web UI (commented out as there is no good validator so far).
+     * 5.  Verify the Jenkins configuration against the schema (commented out as there is no good validator so far).
+     * 6.  Check the Jenkins configuration is valid via Web UI (used the plugin config so far).
+     * 7.  Apply the Jenkins configuration via Web UI (maybe not needed, but we test it as well).
+     * 8.  Write the Jenkins configuration to $JENKINS_ROOT/jenkins.yaml.
+     * 9.  Restart Jenkins.
+     * 10. Check the {@link #stringInLogExpected()} is set during the restart.
+     *
+     * @throws IOException If an exception is thrown managing resources or files.
+     */
+    @Test
+    public void exportImportRoundTrip() throws IOException {
+        String resourcePath = configResource();
+        String resourceContent = getResourceContent(resourcePath);
+
+        assertNotNull(resourcePath);
+        assertNotNull(resourceContent);
+
+        r.then(step -> {
+            // Configure and validate
+            configureWithResource(resourcePath);
+            assertConfiguredAsExpected(r, resourceContent);
+
+            // Get the full configuration
+            String jenkinsConf = getJenkinsConfViaWebUI();
+
+            // Here we check that the exported configuration has the same information as the YAML used.
+            // hack: But when JCasC is running under test (not normally) the ciphered secrets in the YAML are ciphered
+            // again, so the exported configuration won't be the same never. When running normally it works. When the
+            // bug filed is solved we can uncomment that check. In other way, all the plugins with plain secrets in
+            // their YAMLs will fail in this test. Example: Credentials.
+            //TODO: uncomment when https://issues.jenkins-ci.org/browse/JENKINS-57863 is solved
+            assertJenkinsHasConfiguredPath(jenkinsConf, resourcePath);
+
+            //hack: the full Jenkins config fails due to defaultProperties of maven, we use the config of the plugin,
+            //but first, we guarantee that the exported configuration is the same as the configured.
+            //TODO: remove when https://issues.jenkins-ci.org/browse/JENKINS-57122 is solved
+            //@Issue("JENKINS-57122")
+            jenkinsConf = getResourceContent(resourcePath);
+
+            // Get the schema
+            //String schema = getSchemaViaWebUI();
+
+            // Verify it's compliant to the schema
+            // TODO: when YAML schema validation is mature enough we can do that.
```

Comment:
I think we can remove all the commented out code, make's it more difficult to see what's actually being done in the present.
YAML schema is a long way off working its current state

Reply:
ok